core/src/test/kotlin/fr/sncf/osrd/utils/DistanceRangeMapImplTest.kt

```diff
@@ -0,0 +1,214 @@
+        map.putMany(entries)
+        assertEquals("A", map.get(2.5.meters))
+        assertEquals("B", map.get(7.5.meters))
+    }
+
+    @Test
+    fun iterator() {
+        val map = DistanceRangeMapImpl<String>()
+        map.put(0.0.meters, 10.0.meters, "A")
+        val iterator = map.iterator()
+        assertTrue(iterator.hasNext())
+        val entry = iterator.next()
+        assertEquals(0.0.meters, entry.lower)
+        assertEquals(10.0.meters, entry.upper)
+        assertEquals("A", entry.value)
+    }
+
+    @Test
+    fun asList() {
+        val map = DistanceRangeMapImpl<String>()
+        map.put(0.0.meters, 10.0.meters, "A")
+        val list = map.asList()
+        assertEquals(1, list.size)
+        val entry = list[0]
+        assertEquals(0.0.meters, entry.lower)
+        assertEquals(10.0.meters, entry.upper)
+        assertEquals("A", entry.value)
+    }
+
+    @Test
+    fun lowerBound() {
+        val map = DistanceRangeMapImpl<String>()
+        map.put(0.0.meters, 10.0.meters, "A")
+        assertEquals(0.0.meters, map.lowerBound())
+    }
+
+    @Test
+    fun upperBound() {
+        val map = DistanceRangeMapImpl<String>()
+        map.put(0.0.meters, 10.0.meters, "A")
+        assertEquals(10.0.meters, map.upperBound())
+    }
+
+    @Test
+    fun truncate() {
+        val map = DistanceRangeMapImpl<String>()
+        map.put(0.0.meters, 10.0.meters, "A")
+        map.truncate(2.0.meters, 8.0.meters)
+        assertNull(map.get(1.0.meters))
+        assertEquals("A", map.get(5.0.meters))
+        assertNull(map.get(9.0.meters))
+    }
+
+    @Test
+    fun shiftPositions() {
+        val map = DistanceRangeMapImpl<String>()
+        map.put(0.0.meters, 10.0.meters, "A")
+        map.shiftPositions(5.0.meters)
+        assertEquals("A", map.get(10.0.meters))
+    }
+
+    @Test
+    fun get() {
+        val map = DistanceRangeMapImpl<String>()
+        map.put(0.0.meters, 10.0.meters, "A")
+        assertEquals("A", map.get(5.0.meters))
+        assertNull(map.get(15.0.meters))
+    }
+
+    @Test
+    fun subMap() {
+        val map = DistanceRangeMapImpl<String>()
+        map.put(0.0.meters, 10.0.meters, "A")
+        val subMap = map.subMap(2.0.meters, 8.0.meters)
+        assertEquals("A", subMap.get(5.0.meters))
+        assertNull(subMap.get(1.0.meters))
+        assertNull(subMap.get(9.0.meters))
+    }
+
+    @Test
+    fun isEmpty() {
+        val map = DistanceRangeMapImpl<String>()
+        assertTrue(map.isEmpty())
+        map.put(0.0.meters, 10.0.meters, "A")
+        assertFalse(map.isEmpty())
+    }
+
+    @Test
+    fun updateMapIntersection() {
+        val map = DistanceRangeMapImpl<String>()
+        map.put(0.0.meters, 10.0.meters, "A")
+        val updateMap = DistanceRangeMapImpl<String>()
+        updateMap.put(5.0.meters, 15.0.meters, "B")
+        map.updateMapIntersection(updateMap) { old, new -> old + new }
+        assertEquals("AB", map.get(7.5.meters))
```

Comment:
Add the following to test that only the intersection is changed, and the values that are not intersecting in the update map are not added:
`assertEquals("AB", map.get(7.5.meters))`
`assertNull(map.get(12.5.meters))`